setup.py

```diff
@@ -19,6 +19,7 @@ def get_version() -> str:
     "typing-extensions>=3.7.4.3",  # to be able to import TypeAlias
     "importlib_metadata;python_version<'3.8'",
     "packaging>=20.9",
+    "Jinja2",
```

Comment:
I'd say jinja templates are what we should use for modelcards:
- Users can define their own template and pass it to these tools, and it's a well documented format which can easily be extended.
- we're adding html and js code in the modelcard, and HTML is not a [regular language](https://en.wikipedia.org/wiki/Regular_language) and therefore [cannot be parsed with a regex](https://stackoverflow.com/questions/1732348/regex-match-open-tags-except-xhtml-self-contained-tags/1732454#1732454)

But I completely agree that jinja is probably not the best dependency for hfh, and that's partly why I've advocated for modelcards to be its own project/repo rather than being inside hfh.

Reply:
Friendly ping on this to reach a final consensus on this

Reply:
let's add it as optional dependency i.e. only install it when doing it `pip install -e .[cards]` or equivalent?

Reply:
Yes, if adding jinja2 to the `extras_require` is an option, then I'd be more than happy to have this merged in. This extra requirement can be required by the downstream libraries that install `huggingface_hub` in order to manage their model cards, so it's a good compromise for me.